### ISSUE TITLE
Gradle distribution download progress should be visible upon project import

### ISSUE THREAD

**Issue by donat** [eclipse/buildship#351]
Original post: https://bugs.eclipse.org/bugs/show_bug.cgi?id=486627

**Comment by donat** [comment]
@oehme This is still a problem. IIUC Builship should receive the wrapper download events via `org.gradle.tooling.events.ProgressListener`. This doesn't seem to be the case; the first event comes after the wrapper download is finished.

**Comment by oehme** [comment]
Yes, you should be getting a [start and end](https://github.com/gradle/gradle/blob/2f94bfed05d248447439febefb45e841b7d73793/subprojects/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/DistributionInstaller.java#L89) event from the tooling API.

**Comment by badsyntax** [comment]
hi @oehme, I also cannot add a progress listener for this operation. My scenario is the following, which triggers the distribution download:

```
BuildEnvironment environment = connection.model(BuildEnvironment.class).get();
```

I do this before I'm able to add a progress listener to the (`ConfigurableLauncher`) build, and I don't see a way to add a progress listener to the distribution installer. Can you advise on a solution here?

**Comment by badsyntax** [comment]
Sorry, download progress works, not sure why I didn't try this:

```java
ModelBuilder<BuildEnvironment> buildEnvironment = connection.model(BuildEnvironment.class);
buildEnvironment.addProgressListener(
  (ProgressEvent event) -> {
    // event 
  }
});
BuildEnvironment environment = buildEnvironment.get();
```
